gobblin-core-base/src/main/java/gobblin/crypto/EncryptionConfigParser.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package gobblin.crypto;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+
+
+/**
+ * Extract encryption related information from taskState
+ */
+public class EncryptionConfigParser {
+  private static final String ENCRYPT_PREFIX = ".encrypt";
+
+  // TODO these are dupe with ConfigurationKeys - this is to eventually allow for encryption plugins down the road
+  // that just get handed a set of config and can decide to define their own config below the branch prefix if
+  // they want
+  public static final String ENCRYPTION_ALGORITHM_KEY = "algorithm";
+  private static final String ENCRYPTION_KEYSTORE_PATH_KEY = "keystore_path";
+  private static final String ENCRYPTION_KEYSTORE_PASSWORD_KEY = "keystore_password";
+
+  public static final String ENCRYPTION_TYPE_ANY = "any";
+
+  public static Map<String, Object> getConfigForBranch(State taskState, int numBranches, int branch) {
+    String prefixName = ConfigurationKeys.WRITER_PREFIX + ENCRYPT_PREFIX;
+    Map<String, Object> properties = extractPropertiesForBranch(taskState.getProperties(), prefixName,
+        numBranches, branch);
+    if (properties.isEmpty()) {
+      return null;
+    }
+
+    return properties;
+  }
+
+  public static String getEncryptionType(Map<String, Object> properties) {
+    return (String)properties.get(ENCRYPTION_ALGORITHM_KEY);
+  }
+
+  public static String getKeystorePath(Map<String, Object> properties) {
+    return (String)properties.get(ENCRYPTION_KEYSTORE_PATH_KEY);
+  }
+
+  public static String getKeystorePassword(Map<String, Object> properties) {
+    return (String)properties.get(ENCRYPTION_KEYSTORE_PASSWORD_KEY);
```

Comment:
passwords in passwords in passwords :) But yes, will update - didn't realize PasswordManager existed